### PR TITLE
Create controller specs

### PR DESCRIPTION
**Pull request by Tshobohwa**
# FEATURES ADDED:
In this milestone, I created rspec tests for the index and the show methods in the users and posts controller.
For each method, I tested if:
- The response status was correct,
- The correct template is rendered
- The response body includes the correct template text.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb
Comment (KanzaTahreem) [inline spec/controllers/posts_controller_spec.rb:1]: - [ ] Great job on testing the controllers! 👏👍However, as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Create%20a%20Request%20spec%20file%20for%20all%20your%20controllers.),  please ensure that you create a Request spec file for testing all controllers. This means that user and post specs are located in the `spec/requests/` directory and have the `type: request`.  Alternatively, you can use `config.infer_spec_type_from_file_location!` in the RSpec configfile. In this case, simply placing a file in the `spec/requests` directory will suffice.📌You can also refer to the following articles to refresh your knowledge of testing controllers and to learn more about **_Request specs_**. ℹ  Kindly implement this for both controllers.

- [How to test your controllers](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controller_test_lesson.md)
- [Request specs](https://rspec.info/features/6-0/rspec-rails/request-specs/request-spec/#:~:text=Request%20specs%20provide%20a%20thin,if%20you%20have%20set%20config.)

image

  
**_After this implementation, please make sure that all tests are passing before asking for a review_**

_Please feel free to reach out to me on slack if you need any help or further clarification_

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    end
+
+    it 'renders the index template' do
+      get :index, params: { user_id: 1 }
+      expect(response).to render_template(:index)
+    end
+    it 'includes the template content in the response body' do
+      get :index, params: { user_id: 1 }
+      expect(response.body).to include(response.body)
+    end
```
Comment (KanzaTahreem) [inline spec/controllers/posts_controller_spec.rb]: - [ ]  Kindly note that  checking if `response.body` includes itself is always true. Currently, you're checking if the `response.body` includes itself. In other words, it's like asking, "Does this string contain itself?" The answer is always yes. To fix this, kindly check if the response body includes the correct placeholder text that you expect to find. For example:

```ruby
expect(response.body).to include('correct_placeholder_text')
```

**_Please make sure to fix this for both actions(`index`, `show`) of both controllers(`Users`, `Posts`)_**

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @Tshobohwa 

Good job so far👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Both actions are tested
- ✅ Descriptive PR and README file
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Tshobohwa,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**